src/coreclr/vm/gcenv.ee.common.cpp

```diff
@@ -137,6 +137,11 @@ inline bool SafeToReportGenericParamContext(CrawlFrame* pCF)
     return true;
 }
 
+thread_local void **doubleReportTracking = NULL;
```

Comment:
This memory is going to leak.

Can this be allocated/freed in `ScanStackRoots` and passed around in the stack scanning context instead?

Reply:
The intention was to not to have to keep allocating and freeing the array all the time, potentially fragmenting native heap. The leak in the case a thread exits seemed ok for the purpose of this kind of debugging tool. But I can change it to either the way you've mentioned or change the pointer to a holder that would delete it when the thread exits.

Reply:
Short lived allocations do not fragment native heap.

Also, every modern native heap implementation has a thread local cache that the repeated allocation would be served from.